src/Network/OAuth/OAuth2/Internal.hs

```diff
@@ -63,12 +68,40 @@ instance FromJSON OAuth2Token where
 instance ToJSON OAuth2Token where
     toEncoding = genericToEncoding defaultOptions { fieldLabelModifier = camelTo2 '_' }
 
+-- Resources Access Responses https://tools.ietf.org/html/rfc6749#section-7.2
+type ResourceAccessErrors = Void
```

Comment:
slightly prefer to remove it or just more comments to clarify. an unused void type will be a little confusing.